packages/salesforcedx-vscode-core/package.json

```diff
@@ -69,6 +71,12 @@
           "when": "view == sfdx.force.tasks.view"
         }
       ],
+      "explorer/context": [
+        {
+          "command": "sfdx.force.apex.class.create",
+          "when": "explorerResourceIsFolder"
```

Comment:
Using a built-in when clause by VSCode to only show command option when right clicking on a folder in explorer.